src/improvements/tasks/MultisigTask.sol

```diff
@@ -793,26 +810,42 @@ abstract contract MultisigTask is Test, Script, ITask {
         vm.stopPrank();
 
         /// roll back all state changes made during the task
-        require(vm.revertTo(_startSnapshot), "failed to revert back to snapshot, unsafe state to run task");
+        require(
+            vm.revertTo(_startSnapshot), "MultisigTask: failed to revert back to snapshot, unsafe state to run task"
+        );
 
         _processStateDiffChanges(accountAccesses);
 
+        /// there should be at least one account access
+        require(accountAccesses.length > 0, "MultisigTask: no account accesses found");
+
+        /// get the minimum depth of the calls, we only care about the top level calls
+        /// this is to avoid counting subcalls as actions.
+        /// the account accesses are in order of the calls, so the first one is always the top level call
+        uint256 topLevelDepth = accountAccesses[0].depth;
+
         for (uint256 i = 0; i < accountAccesses.length; i++) {
             /// store all gnosis safe storage accesses that are writes
             for (uint256 j = 0; j < accountAccesses[i].storageAccesses.length; j++) {
-                if (accountAccesses[i].account == multisig && accountAccesses[i].storageAccesses[j].isWrite) {
+                if (accountAccesses[i].account == parentMultisig && accountAccesses[i].storageAccesses[j].isWrite) {
                     _accountAccesses.push(accountAccesses[i].storageAccesses[j]);
                 }
             }
 
-            /// only care about top level calls from the multisig,
-            /// static calls are ignored,
             /// calls to and from Addresses and the vm contract are ignored
-            /// ignore calls to vm in the build function
+            bool accountIsNotAddressesOrVm =
+                accountAccesses[i].account != address(addresses) && accountAccesses[i].account != address(vm);
+            bool accessorIsNotAddresses = accountAccesses[i].accessor != address(addresses);
```

Comment:
nit for a future PR: rename the `addresses` var to `addrRegistry` everywhere, the vaguely named `addresses` keeps confusing me lol

Reply:
update in new PR that builds on this one at commit: https://github.com/ethereum-optimism/superchain-ops/pull/606/commits/672353b54374b0aa5605825ad4f8d6d074ddfe0e

Reply:
quick follow up here to no rename imports.